ingress/controllers/gce/controller/utils.go

```diff
@@ -40,6 +40,8 @@ import (
 const (
 	allowHTTPKey    = "kubernetes.io/ingress.allow-http"
 	staticIPNameKey = "kubernetes.io/ingress.global-static-ip-name"
+	ingressClassKey = "kubernetes.io/ingress.class"
```

Comment:
customer using kubectl, or at creation time. They set it when they create an Ingress that they'd like, eg: nginx to use. Running 2 of any controller (rc, endpoint etc) will cause to confusion unless they take a lock, and this is essentially the customer taking the lock on ingress creation.